src/utils/hooks/useConfirmModal.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { useState } from 'react'
+import { Button, Header, Icon, Modal, ModalProps } from 'semantic-ui-react'
+import { useLanguageProvider } from '../../contexts/Localisation'
+
+interface ConfirmModalState extends ModalProps {
+  type: 'warning' | 'confirmation'
+  title: string
+  message?: string
+  OKText: string
+  cancelText: string
+  onOK: () => void
+  onCancel: () => void
+}
+
+const useConfirmModal = ({
```

Comment:
Hmm, i'll comment about this in upstream PR (in omSupply we use global confirm modal and context)

Reply:
I've made a generic useConfirmModal in the next PR, cos this was annoying me too.

Reply:
Oh sorry, this is it. The next PR refactors existing to use it.